### DISCUSSION TITLE
pagview设置composition后，可以给对应的composition动态添加pag文件吗

### DISCUSSION THREAD

**Discussion by camp1990** [Tencent/libpag#1633] [Q&A] [answered]
在pagview或pagimageView设置composition后，可以给对应的composition动态添加pag文件吗，这样可以同时在一个pagview上播放不同的pag动画

**Comment by jhforestever** [comment]
composition是可以叠加的多个pag文件的。

**Reply by camp1990** [reply]
可以在composition动态添加吗，在上一个pag文件播放过程中添加其他pag文件，同时播放多个互不影响

**Comment by jhforestever** [comment]
可以中途切换pag文件,composition .swapLayer()，但可能不是你想要的效果。我觉得可能还是直接让设计师将两个pag文件合成一个好了。

**Reply by camp1990** [reply]
目前有个场景：点击按钮播放动效，快速点击会播放多个且互不影响，如果创建多个pagview会有内存消耗，所以想知道能不能在同一个pagview在composition上动态添加动效layer播放

**Comment by kevingpqi123** [comment, marked as answer]
可以通过 setComposition 的方法不断切换 pag 文件，为了防止卡顿，切换之前先调用 PAGView 的 stop 方法然后再切换
